### ISSUE TITLE
Cannot set vocabulary size for categorical set

### ISSUE THREAD

**Issue by StoyanStAtanasov** [tensorflow/decision-forests#62]
When trying to set the vocabulary size for categorical set column
```python
sentenceFeature = tfdf.keras.FeatureUsage(
    name="sentence", semantic=tfdf.keras.FeatureSemantic.CATEGORICAL_SET, max_vocab_count=10000)
```
I get the error
```
usr/local/lib/python3.6/dist-packages/tensorflow_decision_forests/keras/core.py in __init__(self, name, semantic, discretized, max_vocab_count)
    178       if max_vocab_count is not None:
    179         raise ValueError(
--> 180             "\"max_vocab_count\" only works for CATEGORICAL semantic.")
```

Rationale: I want to train a DF with a text column. This is demonstrated in the intermediate colab example. There I noticed a lot of out of dictionary items. I wanted to increase the dictionary to try to improve the accuracy.
```
CATEGORICAL_SET: 1 (50%)
	0: "sentence" CATEGORICAL_SET has-dict vocab-size:2001 num-oods:12816 (19.0292%) most-frequent:"the" 27205 (40.3941%)
```

This looks to be a bug as in the code to set the vocab_size we allow CATEGORY_SET as well. https://github.com/tensorflow/decision-forests/blob/d530fd85eccc79b47712b8728c9c06f263191a9a/tensorflow_decision_forests/keras/core.py#L192-L197
 Only the error message is the problem
https://github.com/tensorflow/decision-forests/blob/d530fd85eccc79b47712b8728c9c06f263191a9a/tensorflow_decision_forests/keras/core.py#L178-L181

**Comment by StoyanStAtanasov** [comment]
|Category set Dictionary Size|Time|Accuracy|Eval Accuracy|
|---|---|---|---|
|1000|0:40|accuracy:0.767227 logloss:4.58354|accuracy: 0.7271|
|2000|1:40|accuracy:0.806738 logloss:2.41777 |accuracy: 0.7546|
|4000|5:32|accuracy:0.861572 logloss:0.488691|accuracy: 0.8028|

**Comment by achoum** [comment]
Hello StoyanStAtanasov,

Thanks for the detailed report :). Also happy to see that the categorical-set splitter works for you.

This situation was resolved in [this commit](https://github.com/tensorflow/decision-forests/commit/305d7e35f267cad419779911b76cc731fb80a11e) and published in the TF-DF 0.2.0 release.

The following code should run fine:

```python
sentenceFeature = tfdf.keras.FeatureUsage(
  name="sentence",
  semantic=tfdf.keras.FeatureSemantic.CATEGORICAL_SET,
  max_vocab_count=10000)
```

In addition, a global hyper-parameter `max_vocab_count` was also introduced to control the `max_vocab_count` of all the feature usages easily:

```python
model = tfdf.keras.GradientBoostedTreesModel(max_vocab_count=10000)
```

Cheers,
Mathieu

Ps: Have you tried to tune the `categorical_set_split_greedy_sampling` hyper-parameter e.g. try with `categorical_set_split_greedy_sampling=0.2` ?

**Comment by StoyanStAtanasov** [comment]
@achoum Thanks for the fix! Sorry I did not have time to play with categorical_set_split_greedy_sampling.
Cheers,
